### ISSUE TITLE
3D Milling :P (Sorry Todd) 

### ISSUE THREAD

**Issue by openhardwarecoza** [LaserWeb/LaserWeb4#372]
Related to https://plus.google.com/+ArielYahni/posts/iCjFq7t62aK

I remember @tbfleming 's work from months ago (offsetMesh https://tbfleming.github.io/OffsetMesh/ and also some work he did in CP) 

Theres also this project : https://github.com/damianmoore/part-mill by @damianmoore (if we need hands on deck, we can ask (;) 

And of course @nraynaud (who i borrowed Drag Knife maths from for lw3) has http://nraynaud.github.io/webgcode/webapp/visucamTest.html#/ which can also do a little 3d it seems? 

(; now it's an open issue. I say lets start a fundraiser and buy the dev team some OpenBuilds MiniMills (; (+-$500 per person x 3 devs) @cojarbi (we had to use Vectric for https://plus.google.com/+Openbuildspartstore/posts/Epqc6YD7Pgs)

**Comment by nraynaud** [comment]
not to be bragging but webgcode has the most complete 3D milling system of the bunch. In particular because there are various tool shapes supported.

**Comment by openhardwarecoza** [comment]
@raynaud any chance we can bribe you into coding for us? I am sure we can
organise you a mill or a laser?

**Comment by nraynaud** [comment]
now, we're talking :)
To be fair, I'd love me some gear, but 
1) I'm moving continent next month and will be without a permanent place for a while, 
2) I'm an independent contractor, and if I receive "donations" in exchange for work, the tax man might want to have a quick chat with me.

On the serious side: for now I'm committed to other projects, and I'm afraid that telling that I will help you and then not walking the walk would be worst that saying no now.

But I understand that the 3D milling code from webgcode is hard to follow. 

I left some testing code: https://github.com/nraynaud/webgcode/blob/gh-pages/various_tests/test_3D.html ( http://nraynaud.github.io/webgcode/various_tests/test_3D.html )
I think it could help by limiting the scope of what to copy/paste in webgcode (open the example and look at the list of source files in chrome debugger).

Nico

**Comment by nraynaud** [comment]
http://nraynaud.github.io/webgcode/various_tests/test_3D_conservative_rendering.html here the demonstration of the conservative rendering, it show, that the tool (the little squares) will never violate the net shape of the part (the triangles), and the red vertices show how the conservatism is computed.

**Comment by openhardwarecoza** [comment]
Cool man! I appreciate it! Will wait for @tbfleming to look things over
then someday (:

On Jul 6, 2017 1:20 PM, "Nicolas Raynaud" <notifications@github.com> wrote:

http://nraynaud.github.io/webgcode/various_tests/test_
3D_conservative_rendering.html here the demonstration of the conservative
rendering, it show, that the tool (the little squares) will never violate
the net shape of the part (the triangles), and the red vertices show how
the conservatism is computed.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/LaserWeb/LaserWeb4/issues/372#issuecomment-313368524>,
or mute the thread

**Comment by openhardwarecoza** [comment]
@raynaud when you get settled in, feel free to jump back in and ask for
Push rights anytime you want!

**Comment by openhardwarecoza** [comment]
@tbfleming anything useful in the above that could make your life easier?

**Comment by tbfleming** [comment]
I don't know. It takes time to evaluate existing code. I have a strong suspicion that in the 3D cam space, "use X", where X is something you don't like mentioned, will be the best option without years of work.

**Comment by cojarbi** [comment]
Who is this X that we don't like being mentioned ( sound like Harry Potter movie or Beetlejuice )

**Comment by openhardwarecoza** [comment]
(; i assume webasm? (;

Dont mind me, i'm so out of the project my opinion doesn't matter. Dont let
my unwillingness to learn new tech, get in the way of this project (;. I
say Go for it!

**Comment by tbfleming** [comment]
No, I already started on moving toward wasm. Too many performance complaints to stick with pure JS for polygon ops.

X = Fusion 360

**Comment by cojarbi** [comment]
@openhardwarecoza your opinion does matter!

**Comment by openhardwarecoza** [comment]
Ahh ok. Yeah i dont mind it that much. But as a disgruntled Eagle user who
just had to fork over the Premium licence fee... I dont love Autodesk too
much

**Comment by openhardwarecoza** [comment]
Well, closing for now (:
